lib/logdna/client.rb

```diff
@@ -11,153 +12,114 @@ def initialize(request, uri, opts)
       @uri = uri
 
       # NOTE: buffer is in memory
-      @buffer = StringIO.new
-      @messages = []
-      @buffer_over_limit = false
+      @buffer = []
+      @buffer_byte_size = 0
 
-      @side_buffer = StringIO.new
       @side_messages = []
 
       @lock = Mutex.new
-      @task = nil
-
-      # NOTE: the byte limit only affects the message, not the entire message_hash
-      @actual_byte_limit = opts[:flushbyte] ||= Resources::FLUSH_BYTE_LIMIT
-      @actual_flush_interval = opts[:flushtime] ||= Resources::FLUSH_INTERVAL
-
-      @@request = request
+      @flush_limit = opts[:flush_size] ||= Resources::FLUSH_BYTE_LIMIT
```

Comment:
I'm pretty sure we're misusing `||=` here. That operator sets the variable if it's falsy. So what we're doing here is setting `opts[:flush_size]` to `Resources::FLUSH_BYTE_LIMIT` if it's not already set, then setting `@flush_limit` to that value. I don't see the `opts` being used again, I think we just want to set `@flush_limit`, so `@flush_limit = opts[:flush_size] ||= Resources::FLUSH_BYTE_LIMIT`. Same below in line 22 and anywhere else we're using `||=`.